google-cloud-clients/google-cloud-websecurityscanner/src/test/java/com/google/cloud/websecurityscanner/it/v1beta/VpcNegativeTest.java

```diff
@@ -0,0 +1,145 @@
+  private static final String OUT_VPCSC_HOSTNAME = System.getenv(OUT_VPCSC_HOSTNAME_ENV);
+  private static final String OUT_VPCSC_GOOGLE_CREDENTIAL =
+      System.getenv(GOOGLE_CREDENTIAL_DEFAULT_ENV);
+
+  private String testScanConfigCreationDisplayName;
+
+  private static boolean isNotEmpty(String value) {
+    return value != null && value.length() != 0;
+  }
+
+  @BeforeClass
+  public static void setUpClass() {
+    assumeTrue(
+        "To run tests, GOOGLE_CLOUD_TESTS_IN_VPCSC environment variable needs to be set to true",
+        IN_VPCSC_TEST != null && IN_VPCSC_TEST.equalsIgnoreCase("true"));
+
+    assertTrue(
+        OUT_VPCSC_PROJECT_ENV
+            + " environment variable needs to be set to a GCP "
+            + "project that is out of the VPC perimeter",
+        isNotEmpty(OUT_VPCSC_PROJECT));
+
+    assertTrue(
+        OUT_VPCSC_HOSTNAME_ENV + " must be set to a webapp that resides in " + OUT_VPCSC_PROJECT,
+        isNotEmpty(OUT_VPCSC_HOSTNAME));
+
+    assertTrue(
+        GOOGLE_CREDENTIAL_DEFAULT_ENV
+            + " must be set to google application credentials "
+            + "that is outside VPCSC perimeter",
+        isNotEmpty(OUT_VPCSC_GOOGLE_CREDENTIAL));
+  }
+
+  @Before
+  public void setup() {
+    DateTimeFormatter formatter =
+        DateTimeFormat.fullDateTime().withZone(DateTimeZone.forID("America/Los_Angeles"));
+    String currentTime = formatter.print(DateTime.now(DateTimeZone.forID("America/Los_Angeles")));
+    testScanConfigCreationDisplayName = "vpcsc-neg-test" + currentTime + "-0";
+  }
+
+  private WebSecurityScannerSettings getWssSettingWithCredentials() throws IOException {
+    GoogleCredentials credentials =
+        GoogleCredentials.fromStream(new FileInputStream(OUT_VPCSC_GOOGLE_CREDENTIAL))
+            .createScoped(Lists.newArrayList(GOOGLE_API_CLOUD_PLATFORM_LINK));
+    return WebSecurityScannerSettings.newBuilder()
+        .setCredentialsProvider(FixedCredentialsProvider.create(credentials))
+        .build();
+  }
+
+  @Test
+  public void testScanConfigCreation() throws IOException {
+    WebSecurityScannerSettings wssSetting = getWssSettingWithCredentials();
+    final String formattedParent = ProjectName.format(OUT_VPCSC_PROJECT);
+    try (WebSecurityScannerClient wssClient = WebSecurityScannerClient.create(wssSetting)) {
+      ScanConfig scanConfig =
+          ScanConfig.newBuilder()
+              .addAllStartingUrls(Lists.newArrayList(OUT_VPCSC_HOSTNAME))
+              .setDisplayName(testScanConfigCreationDisplayName)
+              .build();
+      CreateScanConfigRequest request =
+          CreateScanConfigRequest.newBuilder()
+              .setParent(formattedParent)
+              .setScanConfig(scanConfig)
+              .build();
+
+      ScanConfig responseScanConfig = wssClient.createScanConfig(request);
+      fail("Exception must occur for non-accessible project");
+    } catch (Exception e) {
```

Comment:
Is there a better exception that we can catch here? Also, add a comment or rename the caught exception the "expected". https://google.github.io/styleguide/javaguide.html#s6.2-caught-exceptions

Reply:
I changed the exception to PermissionDeniedException and renamed it to "expected".